gui/wxpython/mapdisp/main.py

```diff
@@ -370,7 +370,7 @@ def __next__(self):
         return result
 
     def next(self):
-        return self.__next__()
+        return next(self)
```

Comment:
Isn't that a kind of circle?

```
def next(self):
    return next(self)
```

What would be the point of the definition?

Reply:
I'll try to write a better suggestion for it, as you're right, it is confusing, and you make me have second thoughts. Internet  searches seem to point at Python 2/3 compatibility. `next()` is a built in function too, like `len()`. I'm still curious to see in a debugger what is actually called at that point.


The following were useful:


https://stackoverflow.com/questions/28661014/implementing-next-in-a-custom-class

I didn't get to the bottom of it yet

https://stackoverflow.com/questions/19151/how-to-build-a-basic-iterator

https://developer.ibm.com/articles/l-pycon/ (Python 2.1-2.2+)

Reply:
I think it may go up to the weekend to finish sorting this out.

Reply:
It seems to me that "LayerList" class is an iterator, thus the method `next()` may be removed (`__next()__` will be called by default).

Reply:
I'm still not able to trigger this code path. I'm able to get to some functions through d.mon and other display commands, even at the init of LayerList, but I can't reach `def __next__(self):` or `def next(self):`. The closest I have is pylance's docstring/types analysis (vscode default), that points that the second next is the builtin function that would call `__next__`

![image](https://github.com/user-attachments/assets/c8f3d337-f63e-46aa-963c-381821fffb87)
![image](https://github.com/user-attachments/assets/6dc332b6-cff6-4df6-ac9b-5922d281c362)
![image](https://github.com/user-attachments/assets/3f7b6819-72a5-4641-a1b3-034b386539fb)


As such, it really seems like an iterator, and the changes here looks like a Python 2/3 compatibility pattern

https://github.com/echoix/grass/commit/cdb2ad31bf56173e6c7782600664c9ece63bd14b#diff-797e13b9395a539f6439e3f3aacde1699ee78a1152df8ff9590d40378a870167R350-R367

So, @nilason's suggestion makes the most sense

Reply:
Since I can't be sure if it is used in the wild, I kept the `.next()` to call `next()`.